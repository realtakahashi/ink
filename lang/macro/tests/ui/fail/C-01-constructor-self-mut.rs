use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(constructor)]
        fn self_mut_arg(&mut self) -> Self {
            Self {}
        }

        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
