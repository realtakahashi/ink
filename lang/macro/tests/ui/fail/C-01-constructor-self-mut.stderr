error: #[ink(constructor)] functions must not have any kind of `self` receiver
  --> $DIR/C-01-constructor-self-mut.rs:10:25
   |
10 |         fn self_mut_arg(&mut self) -> Self {
   |                         ^^^^^^^^^
